sdk/storage/Azure.Storage.Blobs/src/Sas/BlobSasBuilder.cs

```diff
@@ -324,75 +327,85 @@ private void EnsureState()
+        public override int GetHashCode() =>
+            this.BlobName.GetHashCode() ^
+            this.CacheControl.GetHashCode() ^
+            this.ContainerName.GetHashCode() ^
+            this.ContentDisposition.GetHashCode() ^
+            this.ContentEncoding.GetHashCode() ^
+            this.ContentLanguage.GetHashCode() ^
+            this.ContentType.GetHashCode() ^
+            this.ExpiryTime.GetHashCode() ^
+            this.Identifier.GetHashCode() ^
+            this.IPRange.GetHashCode() ^
+            this.Permissions.GetHashCode() ^
+            this.Protocol.GetHashCode() ^
+            this.StartTime.GetHashCode() ^
+            this.Version.GetHashCode();
 
         /// <summary>
         /// Check if two BlobSasBuilder instances are equal.
         /// </summary>
         /// <param name="left">The first instance to compare.</param>
         /// <param name="right">The second instance to compare.</param>
         /// <returns>True if they're equal, false otherwise.</returns>
-        public static bool operator ==(BlobSasBuilder left, BlobSasBuilder right) => left.Equals(right);
+        public static bool operator ==(BlobSasBuilder left, BlobSasBuilder right) =>
+            left.Equals(right);
 
         /// <summary>
         /// Check if two BlobSasBuilder instances are not equal.
         /// </summary>
         /// <param name="left">The first instance to compare.</param>
         /// <param name="right">The second instance to compare.</param>
         /// <returns>True if they're not equal, false otherwise.</returns>
-        public static bool operator !=(BlobSasBuilder left, BlobSasBuilder right) => !(left == right);
+        public static bool operator !=(BlobSasBuilder left, BlobSasBuilder right) =>
+            !(left == right);
 
         /// <summary>
         /// Check if two BlobSasBuilder instances are equal.
         /// </summary>
         /// <param name="obj">The instance to compare to.</param>
         /// <returns>True if they're equal, false otherwise.</returns>
-        public bool Equals(BlobSasBuilder other)
-            => this.BlobName == other.BlobName
-            && this.CacheControl == other.CacheControl
-            && this.ContainerName == other.ContainerName
-            && this.ContentDisposition == other.ContentDisposition
-            && this.ContentEncoding == other.ContentEncoding
-            && this.ContentLanguage == other.ContentEncoding
-            && this.ContentType == other.ContentType
-            && this.ExpiryTime == other.ExpiryTime
-            && this.Identifier == other.Identifier
-            && this.IPRange == other.IPRange
-            && this.Permissions == other.Permissions
-            && this.Protocol == other.Protocol
-            && this.StartTime == other.StartTime
-            && this.Version == other.Version
-            ;
+        public bool Equals(BlobSasBuilder other) =>
+            this.BlobName == other.BlobName &&
```

Comment:
Is there a concern with the format of the string where we would need to do a `StringComparison.Ordinal` ? or equivalent?

Reply:
This is a great question.  @kfarmer-msft and @seanmcc-msft  - is this something we care about?

Reply:
I'm tempted to say we probably want exact matches for something like this.  Keep in mind that the server's pretty unforgiving, and constructs its own version of the SAS string-to-sign when it does verification.

Reply:
That sounds good to me.